include/openthread/dataset.h

```diff
@@ -594,6 +594,15 @@ otError otNetworkNameFromString(otNetworkName *aNetworkName, const char *aNameSt
  */
 otError otDatasetParseTlvs(const otOperationalDatasetTlvs *aDatasetTlvs, otOperationalDataset *aDataset);
 
+/**
+ * This function updates an Operational Dataset with information from another Operational Dataset.
+ *
+ * @param[in]  aSrcDataset  A pointer to dataset with the new/updated information.
+ * @param[out] aDstDataset  A pointer to where the dataset will be placed.
+ *
+ */
```

Comment:
Add comment explaining how "update" work for datasets?
```suggestion
/**
 * This function updates an Operational Dataset with information from another Operational Dataset.
 *
 * This function updates @p aDstDataset by setting all the components that are marked as present in @p aSrcDataset to
 * the same value. Other components remain unchanged in @p aDataset.
 * 
 * @param[in]  aSrcDataset  A pointer to dataset with the new/updated information.
 * @param[out] aDstDataset  A pointer to where the dataset will be placed.
 *
 */
```